### ISSUE TITLE
Noisy inference results with TF2 int8 tflite models

### ISSUE THREAD

**Issue by Rahn80643** [ARM-software/armnn#637]
Hi,

I'm migrating our segmentation model from TensorFlow1 to TensorFlow2, and the model is retrained with [TensorFlow quantization aware training technique](https://www.tensorflow.org/model_optimization/guide/quantization/training); however, the inference results of TF2 tflite model from ArmNN are not ideal, the comparisons and environment settings are shown below:

![armnn2202_1](https://user-images.githubusercontent.com/8951991/165028363-4b5485c0-41c4-4557-b2b2-ec91b18a5b12.png)

![armnn2202_2](https://user-images.githubusercontent.com/8951991/165028378-e458af47-ee8a-4167-bb4e-f2dcac24d73d.png)

- ArmNN 22.02
- Compute Library 22.02
- FlatBuffers 1.12.0
- Protobuf 3.12.0
- TensorFlow 2.4.0

Based on the inference results from TensorFlow tflite API, the weights in tflite models seems normal, but the inference results from ArmNN contain many noisy pixels, I want to ask:
1. Is this issue resulted from the version of TensorFlow or tflite?
2. How to effectively improve the inference results in ArmNN?

Best Regards,
Rahn

**Comment by TeresaARM** [comment]
Hi @Rahn80643,

thank you very much for your comment. which backend are you using when you run ArmNN, CpuAcc, GpuAcc or CpuRef?

**Comment by Rahn80643** [comment]
Hi @TeresaARM,

I've tested with CpuAcc and GpuAcc backend, and the inference results are similar

**Comment by TeresaARM** [comment]
Hi @Rahn80643,

could you give it a go to CpuRef? and see if the results are closer to TensorFlow? The performance will be worse than CpuAcc and GpuAcc, as CpuRef is just a "reference", but this information will give help us to find out where the issue is.

Thank you!

**Comment by Rahn80643** [comment]
Hi @TeresaARM,

The followings are the inference results using CpuRef, and the results are not closer to those from TensorFlow

![armnn2202_1](https://user-images.githubusercontent.com/8951991/165432675-0700f2a4-f6c5-47ea-9fa4-124f5c38904a.png)

![armnn2202_2](https://user-images.githubusercontent.com/8951991/165432696-f20a4fbc-e2ce-4e06-a9ba-9928f163c61c.png)

Thank you

**Comment by Rahn80643** [comment]
Hi,

I compared two int8 tflite models trained in TensorFlow1 and TensorFlow2 respectively, and I found the input zero point values are different. Is this the main reason why inferring TensorFlow2 int8 tflite model generates noisy results? If so, how to include the `input zero point` into ArmNN inference code to solve this issue?

![zero_point](https://user-images.githubusercontent.com/8951991/168243269-5aa5f9d3-d088-49a5-a44e-33759714dff3.png)

Rahn

**Comment by MatthewARM** [comment]
yes, this could be the problem. In your application code, when you convert the input image into a tensor for Arm NN, are you taking the zero-point into account?

**Comment by MatthewARM** [comment]
If this is the problem, there is some good explanation of it here: https://github.com/ARM-software/armnn/issues/161

**Comment by Rahn80643** [comment]
Hi,

I've tried the following modifications but the inference results still contain noisy results:
1. Based on thread [#161](https://github.com/ARM-software/armnn/issues/161), the author had tried not to do preprocess the image and divide the output by 255. I tried the same process in ArmNN; however, since the values of output results are within the range of [0, 255], if the values are divided by 255, the results will be all 0s.

2. the input images are preprocessed based on the following formula in ArmNN, 
new_value = ((old_value/scale_factor) - mean)/ stddev. Therefore, I've tried changing mean values, but the inference results still contain noisy results:

![different mean values](https://user-images.githubusercontent.com/8951991/168945478-fb0304bc-613d-4f3a-88ea-ae5423456269.png)

Apart from modifying the mean values in preprocess stage, are there other approaches to take the zero-point into account in ArmNN settings?
